engines/group/app/controllers/group/api/v1/groups_controller.rb

```diff
@@ -5,31 +5,39 @@ class GroupsController < ApplicationController
 
         before_action :authenticate
         before_action :find_and_authorize_group, only: [:show, :update, :destroy]
+        before_action :initialize_side_effects, only: [:create]
+        after_action :add_side_effects_links, only: [:create]
 
+        # GET api/v1/groups
+        #
         def index
           render json: GroupsSerializer.new(current_user.groups)
         end
 
+        # GET api/v1/groups/:id
+        #
         def show
           render json: GroupSerializer.new(@group)
         end
 
+        # POST api/v1/groups
+        #
         def create
           group = ::Group::Group.new(group_params)
-          group_creation = GroupCreation.new(group, current_user)
-          if group_creation.create
+          group_creator = GroupCreator.new(group, current_user)
+          if group_creator.create
+            @side_effects << group_creator.side_effects
             render json: GroupSerializer.new(group)
           else
             render(
               status: :bad_request,
-              json: {
-                model: group.class.name,
-                errors: group.errors.full_messages
-              }
+              json: group.errors.to_json
```

Comment:
This is the format I'm using in Signup Complete. Just an object with errors. No need for type